wallet/src/script.rs

```diff
@@ -0,0 +1,352 @@
+use derive_tag::{FromTag, ToTag};
+use std::convert::TryInto;
+use tag::{FromTag, ToTag};
+
+#[allow(non_camel_case_types)]
+#[derive(Debug, PartialEq, ToTag, FromTag)]
+#[repr(u8)]
+pub enum Script {
+    // holds bytes pushed to the stack and the value of the amount of bytes that
+    // were supposed to be pushed (in little-endian)
+    OP_FALSE = 0x00, // equivalent to OP_0
+    OP_PUSHDATA1 = 0x4c,
+    OP_PUSHDATA2 = 0x4d,
+    OP_PUSHDATA4 = 0x4e,
+    OP_1NEGATE = 0x4f,
+    OP_TRUE = 0x51, // equivalent to OP_1
+    OP_2 = 0x52,
+    OP_3 = 0x53,
+    OP_4 = 0x54,
+    OP_5 = 0x55,
+    OP_6 = 0x56,
+    OP_7 = 0x57,
+    OP_8 = 0x58,
+    OP_9 = 0x59,
+    OP_10 = 0x5a,
+    OP_11 = 0x5b,
+    OP_12 = 0x5c,
+    OP_13 = 0x5d,
+    OP_14 = 0x5e,
+    OP_15 = 0x5f,
+    OP_16 = 0x60,
+
+    OP_NOP = 0x61,
+    OP_IF = 0x63,
+    OP_NOTIF = 0x64,
+    OP_ELSE = 0x67,
+    OP_ENDIF = 0x68,
+    OP_VERIFY = 0x69,
+    OP_RETURN = 0x6a,
+
+    OP_TOALTSTACK = 0x6b,
+    OP_FROMALTSTACK = 0x6c,
+    OP_2DROP = 0x6d,
+    OP_2DUP = 0x6e,
+    OP_3DUP = 0x6f,
+    OP_2OVER = 0x70,
+    OP_2ROT = 0x71,
+    OP_2SWAP = 0x72,
+    OP_IFDUP = 0x73,
+    OP_DEPTH = 0x74,
+    OP_DROP = 0x75,
+    OP_DUP = 0x76,
+    OP_NIP = 0x77,
+    OP_OVER = 0x78,
+    OP_PICK = 0x79,
+    OP_ROLL = 0x7a,
+    OP_ROT = 0x7b,
+    OP_SWAP = 0x7c,
+    OP_TUCK = 0x7d,
+
+    OP_SIZE = 0x82,
+
+    OP_EQUAL = 0x87,
+    OP_EQUALVERIFY = 0x88,
+
+    OP_1ADD = 0x8b,
```

Comment:
It seems that other Arithmetic operations are missing, is this intentional? (Eg: OP_2MUL)